client/modules/IDE/components/FileNode.jsx

```diff
@@ -373,14 +379,24 @@ class FileNode extends React.Component {
                     </li>
                     {this.props.authenticated && (
                       <li>
-                        <button
-                          aria-label={t('FileNode.UploadFileARIA')}
-                          onClick={this.handleClickUploadFile}
-                          onBlur={this.onBlurComponent}
-                          onFocus={this.onFocusComponent}
-                        >
-                          {t('FileNode.UploadFile')}
-                        </button>
+                        {currentSize < sizeLimit ? (
```

Comment:
Maybe this could be handled by changing the css property depending on `currentSize < sizeLimit` rather than the whole `button`?